onecodex/lib/enums.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+
+__all__ = ["Field", "AbundanceField", "AlphaDiversityMetric", "BetaDiversityMetric"]
+
+
+class BaseEnum(str, Enum):
+    @classmethod
+    def has_value(cls, value):
+        return value in cls.values()
+
+    @classmethod
+    def values(cls):
+        return [e.value for e in cls]
+
+
+class AbundanceField(BaseEnum):
+    Abundance = "abundance"
+    AbundanceWChildren = "abundance_w_children"
+
+
+class Field(BaseEnum):
```

Comment:
Should this be renamed to `Metric` and `AbundanceMetric`. "Field" feels very generic. Realize that's a larger API change probably...

Reply:
@clausmith I've updated the has_value calls, i'll let you weight in on the naming. Let me know what you want/prefer and I'll update it.

Reply:
There are one or two other places where we have a kwarg for `field`, but I don't have a strong preference either way. As long as `Metric` wouldn't get confusing with `AlphaDiversityMetric` and `BetaDiversityMetric` since it's distinct then I think we're fine to change `Field` -> `Metric`.